plugins/woocommerce/includes/admin/helper/class-wc-helper.php

```diff
@@ -1676,6 +1646,111 @@ public static function log( $message, $level = 'info' ) {
 
 		self::$log->log( $level, $message, array( 'source' => 'helper' ) );
 	}
+
+	/**
+	 * Handles WC Helper disconnect tasks.
+	 *
+	 * @return void
+	 */
+	public static function disconnect() {
+		WC_Helper_API::post(
+			'oauth/invalidate_token',
+			array(
+				'authenticated' => true,
+			)
+		);
+
+		WC_Helper_Options::update( 'auth', array() );
+		WC_Helper_Options::update( 'auth_user_data', array() );
+
+		self::_flush_subscriptions_cache();
+		self::_flush_updates_cache();
+	}
+
+	/**
+	 * Checks if `access_token` exists in `auth` option.
+	 *
+	 * @return bool
+	 */
+	public static function is_site_connected(): bool {
+		$auth = WC_Helper_Options::get( 'auth' );
+
+		// If `access_token` is empty, there's no active connection.
+		return ! empty( $auth['access_token'] );
+	}
+
+	/**
+	 * Allows to connect with WCCOM using application password. used it to connect via CLI
+	 *
+	 * @param string $password The application password
+	 *
+	 * @return void|WP_Error
+	 */
+	public static function connect_with_password( string $password ) {
+		$request = WC_Helper_API::post(
+			'connect',
+			array(
```

Comment:
The content type should be set to JSON. If not the connect endpoint won't be able to read parameters.

```
array(
		'headers' => array(
			'X-API-Key' => $password,
			'Content-Type'=> 'application/json',
		),
		'body'    => wp_json_encode( array( 'home_url' => home_url() ) ),
	)
```

Reply:
Added in 00e0132